tensorflow/compiler/mlir/tosa/transforms/dequantize_tfl_softmax.cc

```diff
@@ -0,0 +1,91 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "mlir/Dialect/Quant/QuantOps.h"
+#include "mlir/Dialect/Tosa/IR/TosaOps.h"                // from @llvm-project
+#include "mlir/IR/MLIRContext.h"                         // from @llvm-project
+#include "mlir/Pass/Pass.h"                              // from @llvm-project
+#include "mlir/Support/LogicalResult.h"                  // from @llvm-project
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"  // from @llvm-project
+#include "tensorflow/compiler/mlir/tensorflow/ir/tf_ops.h"
+#include "tensorflow/compiler/mlir/tosa/transforms/passes.h"
+
+#define PASS_NAME "tosa-dequantize_tfl-softmax"
+#define DEBUG_TYPE PASS_NAME
+
+namespace mlir {
+namespace tosa {
+namespace {
+
+class TosaDequantizeTFLSoftmax
+    : public TosaDequantizeTFLSoftmaxPassBase<TosaDequantizeTFLSoftmax> {
+ public:
+  explicit TosaDequantizeTFLSoftmax() {}
+  void runOnOperation() override;
+};
+
+struct TosaDequantizeTFLSoftmaxPattern : public RewritePattern {
+  explicit TosaDequantizeTFLSoftmaxPattern(MLIRContext* context)
+      : RewritePattern(TFL::SoftmaxOp::getOperationName(), 1, context) {}
+  LogicalResult matchAndRewrite(Operation* op,
+                                PatternRewriter& rewriter) const override;
+};
+
+LogicalResult TosaDequantizeTFLSoftmaxPattern::matchAndRewrite(
+    Operation* op, PatternRewriter& rewriter) const {
+  fprintf(stderr, "enter matchAndRewrite\n");
+  TFL::SoftmaxOp tfl_softmax_op = cast<TFL::SoftmaxOp>(op);
+  RankedTensorType input_type =
+      tfl_softmax_op.input().getType().cast<RankedTensorType>();
+  if (!input_type.getElementType().isa<mlir::quant::QuantizedType>()) {
+    fprintf(stderr, "not quantized, return failure from matchAndRewrite\n");
```

Comment:
Nit: llvm::errs() stream appears to be preferred to stderr.

Reply:
sorry, meant to drop these printfs. thanks for catching.